services/mfa/mfa.go

```diff
@@ -52,39 +55,33 @@ func getIssuerFromUrl(uri string) string {
 	return url.QueryEscape(issuer)
 }
 
-func (m *Mfa) GenerateSecret(user *model.User) (string, []byte, *model.AppError) {
-	if err := m.checkConfig(); err != nil {
-		return "", nil, err
-	}
+// GenerateSecret generates a new user mfa secret and store it with the StoreSecret function provided
```

Comment:
I am wondering if we can use the consumer interface approach here? Using methods is the right approach when there is shared state involved. And the store certainly seems to be a shared state. The way we are using the package right now is incorrect. We are dynamically creating the struct and calling its methods, whereas we should just create the struct once and store it in `*Server` and call its methods.

Getting rid of the config service seems okay because we only use the siteURL. But for the store, I'd suggest to use a consumer interface so that consumers of the package can pass a proper store interface.

Here is what I would suggest:

```
type Store interface{
// add stuff here
}

type MFA struct{
  store Store
}

func New(store Store) *MFA {
}

func (m *MFA) GenerateSecret(siteURL, userEmail, userID string) (string, []byte, error) {}
func (m *MFA) Activate(userMFASecret, userID string, token string) error {}
func (m *MFA) Deactivate(userID string) error {}
func (m *MFA) ValidateToken(secret, token string) (bool, error) {}
```

Reply:
That was my first approach, but I liked the idea of having just methods, and remove the need of the instantiation of the library itself. Specially because the dependency with the config, but now, if we only depend on the store, makes a lot of sense. Let me rewrite that.

Reply:
done